server.go

```diff
@@ -340,6 +392,173 @@ func (a *App) VerifyHandler(w http.ResponseWriter, r *http.Request) {
 	logRequest(r, 200)
 }
 
+func (a *App) LocksHandler(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	repo := vars["repo"]
+	auth := r.Header.Get("Authorization")
+
+	enc := json.NewEncoder(w)
+	ll := &LockList{}
+
+	w.Header().Set("Content-Type", metaMediaType)
+
+	locks, nextCursor, err := a.metaStore.FilteredLocks(auth, repo,
+		r.FormValue("path"),
+		r.FormValue("cursor"),
+		r.FormValue("limit"))
+
+	if err != nil {
+		ll.Message = err.Error()
+	} else {
+		ll.Locks = locks
+		ll.NextCursor = nextCursor
+	}
+
+	enc.Encode(ll)
+
+	logRequest(r, 200)
+}
+
+func (a *App) LocksVerifyHandler(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	repo := vars["repo"]
+	auth := r.Header.Get("Authorization")
+
+	dec := json.NewDecoder(r.Body)
+	enc := json.NewEncoder(w)
+
+	w.Header().Set("Content-Type", metaMediaType)
+
+	reqBody := &VerifiableLockRequest{}
+	if err := dec.Decode(reqBody); err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		enc.Encode(struct {
+			Message string `json:"message"`
+		}{"json decode error: " + err.Error()})
+		return
+	}
+
+	ll := &VerifiableLockList{}
+	locks, nextCursor, err := a.metaStore.FilteredLocks(auth, repo, "",
+		reqBody.Cursor,
+		strconv.Itoa(reqBody.Limit))
+	if err != nil {
+		ll.Message = err.Error()
+	} else {
+		ll.NextCursor = nextCursor
+
+		for _, l := range locks {
+			if strings.Contains(l.Path, "theirs") {
+				ll.Theirs = append(ll.Theirs, l)
+			} else {
+				ll.Ours = append(ll.Ours, l)
+			}
+		}
+	}
+
+	enc.Encode(ll)
+
+	logRequest(r, 200)
+}
+
+func (a *App) CreateLockHandler(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	user := vars["user"]
+	repo := vars["repo"]
+	auth := r.Header.Get("Authorization")
+
+	dec := json.NewDecoder(r.Body)
+	enc := json.NewEncoder(w)
+
+	w.Header().Set("Content-Type", metaMediaType)
+
+	var lockRequest LockRequest
+	if err := dec.Decode(&lockRequest); err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
```

Comment:
Do you think a status of `400`, `http.StatusBadRequest` would be more appropriate here?